Elly Vending step1

dely2p opened the pull request:
Vending step1 입니다.

dely2p committed:
create VendingMachineApp #1

음료 자판기 아이패드 앱 프로젝트 생성

dely2p committed:
add Level2 VendingMachine classes #1

Level2의 음료자판기 클래스들을 추가함

dely2p committed:
add each kind of Beverages object #3

딸기우유, 바나나우유, 팹시콜라를 생성했다.

dely2p committed:
add code in viewDidLoad() method to addInventory #3

생성한 각 음료수객체를 사용하여 재고 추가를 했다.

dely2p committed:
add SwiftLint

dely2p committed:
add code to showEntireInventory #3

재고목록 출력 구현

dely2p committed:
fix readme.md file to step1 description #2

readme.md file에 실행 콘솔화면 및 코드 작성

godrm commented on VendingMachineApp/VendingMachineApp/BeverageGroup/SodaGroup/Sprite.swift:
```diff
@@ -0,0 +1,25 @@
+    }
+    override var description: String {
+        return "\(self.kindOf)(\(type(of: self)))\(super.description)"
+    }
```
지금 보니 여기 override var description: String { } 모두 중복된 코드네요.
좋은 방법이 있을까요?

dely2p committed:
fix override description property #4

Beverage description property를 Soda, Milk, Coffee에서 상속받고 세부 음료에서도 상속받아서 각각 쓴 것을 Soda, Milk, Coffee에서 다른 property를 생성하여 세부 음료에서 상속받아 사용하도록 개선함

godrm commented:
이거 개선한거 어떻게 검증/테스트 해봤나요?

dely2p commented:
이전에 Level2에서 만든 description이라 VendingMachine main()에서
print(coffee.descriptionDetail) 해서 확인해보았습니다.

이후 음료자판기 앱에서도 확인해보려 breakpoint를 사용했는데
(po strawberryMilk.descriptionDetail)
이전에 VendingMachine과는 달리 재고 추가를 각 음료 타입을 Beverage로 해서 
descriptionDetail에 접근할 수 없는 것을 발견했습니다.
그래서 수정하였습니다.

그런데 혹시 UnitTest를 하는 게 더 나을까요??

godrm commented:
코드를 바꾸고 나면 그걸 검증하는 단계는 항상 필요합니다.
print()는 대안이 아니에요. 항상 테스트 코드를 작성하는 습관을 가지세요.
